### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by ksigei**
### In this milestone, I was able to:

- Remove all hardcoded items from the tasks array.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- By default new tasks have the property completed set to false and the property index set to the value of the new array length 
- Ensure deleting a task updates all remaining items' indexes, so they represent the current list order and are unique.
- Ensure all changes to the To Do List are saved in local storage.

### CODE REVIEW

File: src/modules/local_storage.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+export function addTodo(e) {
+  e.preventDefault();
+
+  createElements(todoInput.value);
+
+  saveTodoInLocalStorage(todoInput.value);
+
+  todoInput.value = '';
+}
```
Comment (Whoistolu) [inline src/modules/local_storage.js]: - [ ] [REQUIRED] From the image below what you have is an array of strings, but what you should have is an array of objects. Each of the task object should have `description` (string), `completed` (boolean), and `index` (number) properties. Kindly refactor accordingly.

  Screenshot from 2022-03-31 19-52-35

- [ ] [REQUIRED] When a task is added the `completed` property should be set to `false` and `index` property should be set to the new array length.

  ```
  [
    {description: 'first task', completed: false, index: 1}
  ]
  ```

File: src/modules/new_todo.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  checkButton.classList.add('fas', 'fa-check', 'check-btn'); // error if use space, so split class with '',
+
+  todoDiv.appendChild(checkButton);
+
+  const deleteButton = document.createElement('button');
+  deleteButton.classList.add('fas', 'fa-trash', 'delete-btn');
+  todoDiv.appendChild(deleteButton);
+
+  todoList.appendChild(todoDiv);
+};
```
Comment (Whoistolu) [inline src/modules/new_todo.js]: - [ ] [REQUIRED] Kindly add a feature and functionality that allows this added task to be edited.

- [ ] [REQUIRED] When a task is edited the `description` property in the localStorage of the particular should also be updated as every user action should be preserved in the localStorage.

File: src/modules/local_storage.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+const removeTodoFromLocalStorage = (item) => {
+  const items = existOrNot();
+
+  const itemIndex = item.children[0].innerText;
+
+  items.splice(items.indexOf(itemIndex), 1);
+
+  localStorage.setItem('items', JSON.stringify(items));
+};
```
Comment (Whoistolu) [inline src/modules/local_storage.js]: - [ ] [REQUIRED] Ensure that after deleting a task the `index` property of each remaining task is also updated to the corresponding position in the array.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @ksigei 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — APPROVED** [review]
Hi @ksigei , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**